### PR TITLE
Copy index order with ModelItems

### LINKED ISSUE #241: Data Class ordering not maintained when creating a new version
When creating a new version of a model, the classes are copied but the ordering is not maintained.  This happened within a versioned folder, but I assume the issue applies on a single model as well.  The `idx` field in the database is set, but not with the correct ordering.

### PR DESCRIPTION
**Pull request by joe-crawford**
- When copying (reference) data classes, elements, types and enumeration values as part of another item, copy their indices (resolves #241)

### CODE REVIEW

File: mdm-plugin-terminology/grails-app/services/uk/ac/ox/softeng/maurodatamapper/terminology/item/TermService.groovy [hunk 1]
```diff
@@ -236,6 +236,13 @@ class TermService extends ModelItemService<Term> {
         copy
     }
 
+    Term copyTerm(CodeSet copiedCodeSet, Term original, User copier, UserSecurityPolicyManager userSecurityPolicyManager,
```
Comment (olliefreeman) [inline mdm-plugin-terminology/grails-app/services/uk/ac/ox/softeng/maurodatamapper/terminology/item/TermService.groovy:239]: so close to not having any comments....

we never copy terms when we copy a codeset as terms are owned by a terminology, when copying a codeset we always use the existing term and just add the existing term to the copied codeset, therefore this method needs to be removed

**Review by olliefreeman — CHANGES_REQUESTED** [review]
Only 1 issue that needs fixing

**Comment by joe-crawford** [comment]
Failing test looks to be unrelated

File: mdm-plugin-terminology/grails-app/services/uk/ac/ox/softeng/maurodatamapper/terminology/item/TermService.groovy (hunk 1 above)
Comment (joe-crawford) [reply mdm-plugin-terminology/grails-app/services/uk/ac/ox/softeng/maurodatamapper/terminology/item/TermService.groovy]: Have removed.

**Pull request merged**